src/time_steppers.jl

```diff
@@ -34,46 +37,32 @@ function time_step!(model::Model{A}, Nt, Δt) where A <: Architecture
     Δx, Δy, Δz = model.grid.Δx, model.grid.Δy, model.grid.Δz
 
     # Unpack model fields
-         grid = model.grid
-        clock = model.clock
-          eos = model.eos
-    constants = model.constants
-            U = model.velocities
-           tr = model.tracers
-           pr = model.pressures
-      forcing = model.forcing
-      closure = model.closure
+              grid = model.grid
+             clock = model.clock
+               eos = model.eos
+         constants = model.constants
+                pr = model.pressures
+           forcing = model.forcing
+           closure = model.closure
     poisson_solver = model.poisson_solver
-
-    bcs = model.boundary_conditions
-    G = model.G
-    Gp = model.Gp
+     diffusivities = model.diffusivities 
+               bcs = model.boundary_conditions
 
     # We can use the same array for the right-hand-side RHS and the solution ϕ.
-    RHS, ϕ = poisson_solver.storage, poisson_solver.storage
-
-    gΔz = model.constants.g * model.grid.Δz
-    fCor = model.constants.f
-
-    uvw = U.u.data, U.v.data, U.w.data
-    TS = tr.T.data, tr.S.data
-    Guvw = G.Gu.data, G.Gv.data, G.Gw.data
-
-    # Source terms at current (Gⁿ) and previous (G⁻) time steps.
-    Gⁿ = G.Gu.data, G.Gv.data, G.Gw.data, G.GT.data, G.GS.data
-    G⁻ = Gp.Gu.data, Gp.Gv.data, Gp.Gw.data, Gp.GT.data, Gp.GS.data
-
+    RHS = poisson_solver.storage
+    U, ϕ, Gⁿ, G⁻ = datatuples(model.velocities, model.tracers, model.G, model.Gp)
+    Guvw = Gⁿ[1:3]
     FT = eltype(grid)
 
     # Field tuples for fill_halo_regions.
-    u_ft = (:u, bcs.u, U.u.data)
-    v_ft = (:v, bcs.v, U.v.data)
-    w_ft = (:w, bcs.w, U.w.data)
-    T_ft = (:T, bcs.T, tr.T.data)
-    S_ft = (:S, bcs.S, tr.S.data)
-    Gu_ft = (:u, bcs.u, G.Gu.data)
-    Gv_ft = (:v, bcs.v, G.Gv.data)
-    Gw_ft = (:w, bcs.w, G.Gw.data)
+    u_ft = (:u, bcs.u, U[1])
+    v_ft = (:v, bcs.v, U[2])
+    w_ft = (:w, bcs.w, U[3])
+    T_ft = (:T, bcs.T, ϕ[1])
```

Comment:
I don't mind indexing by number here: the only special fields are the velocity components, which are naturally indexed by number, as the velocity field is a vector with components/indices x=1, y=2, z=3. 

As for tracers, their names are not relevant except for the calculation of density/buoyancy. I feel the system we want is actually to pass a vector of tracer fields to the density/buoyancy function, which then returns the density/buoyancy perturbation associated with those tracer fields. This abstracts the calculation of density to arbitrary tracer fields. In this system, the names of tracer fields must only be embedded within the tracer vector (as occurs for a `NamedTuple`), which obviates the need to refer to tracers by name in `time_step!`.

Does it seem clear that `U[3] = w`? In my experience the notation `U[3]` is actually more clear and more widely used in fluid dynamics than the name `w` for the third component of the velocity vector.

I think ultimately we will want to distinguish between parts of the time-stepper that depend on physics (like the calculation of interior source terms) from parts that are elements of a generic time-stepping scheme. We need this to generalize to arbitrary numbers of tracers and also to abstract the time-stepping to allow the selection of different time-stepping schemes.

Lastly, this discussion may be moot because I believe we should preform the objects we need for time-stepping in the model constructor rather than performing lots of aliasing and translations inside the `time_step!` function. I think it is important that `time_step!` is compact for readability and maintainability.